### PR TITLE
Create new event frontend issue/17

### PR DESCRIPTION
**Pull request by mefaba**
### What this PR does (required):
- Created a component for Event Creation.
- Created page for new-event-type
- The component that creates event is designed in a way that we can use it later for implementing editing event kind of feature.
### Screenshots / Videos (required):
image
### Any information needed to test this feature (required):
Validation:
image

### Any issues with the current functionality (optional):
- Validation is available but not perfect. Could not trigger error message with touched prop of formik. Because of that right now error messages are displayed even if you are not focused that field.

### CODE REVIEW

File: client/src/components/NewEventType/NewEventType.tsx [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+const NewEventType = ({ initialFormValues = defaultFormValues }: Props): JSX.Element => {
+  const classes = useStyles();
+  const history = useHistory();
+
+  const handleFormSubmit = ({ eventName, eventDescription, eventUrl, eventDuration }: Event) => {
+    createEvent({ eventName, eventDescription, eventUrl, eventDuration }).then((data) => {
+      if (data.success) {
+        history.push('/dashboard');
+      }
+      //TODO catch error
```
Comment (moffatethan) [inline client/src/components/NewEventType/NewEventType.tsx]: I don't understand why not just catch the error? Is there a reason we need to have a TODO here?

**Review by moffatethan — COMMENTED** [review]
Left a comment, also. We should probably do validation on the front-end for the form.

File: client/src/components/NewEventType/NewEventType.tsx (hunk 1 above)
Comment (mefaba) [reply client/src/components/NewEventType/NewEventType.tsx]: I was thinking, maybe we would want to render according to loading/error/data states. I mean we would have something like below:
`const { loading, error, data} = useQuery(apiCallFunction)`

After that we would condionally render. But I forgot to mention this in yesterday's meeting

Comment (mefaba) [reply client/src/components/NewEventType/NewEventType.tsx]: fixed

**Review by moffatethan — APPROVED** [review]
Thank you for doing the changes Fatih! Good to merge

**Pull request merged**